### PR TITLE
Check if provider is supported

### PR DESCRIPTION
**Pull request by tisnik**
## Description

- Fix logging message template - avoid key error
- Unit test checks that key error is not thrown
- Program logic will probably need to be updated, but not in this simple bug fix

## Type of change

- [ ] Refactor
- [ ] New feature
- [x] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library
- [x] Unit test

### CODE REVIEW

File: tests/unit/llms/test_llm_loader.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    # we use bare Exception in the code, so need to check
+    # message, at least
+    with pytest.raises(Exception, match="ERROR: Missing provider"):
+        LLMLoader(provider=None)
+
+
+def test_constructor_wrong_provider():
+    """Test how wrong provider is checked."""
+    # currently, it just logs error and that's it
+    LLMLoader(provider="foo-bar-baz", model=constants.GRANITE_13B_CHAT_V1)
```
Comment (bparees) [inline tests/unit/llms/test_llm_loader.py:20]:
```suggestion
    LLMLoader(provider="invalid-provider", model=constants.GRANITE_13B_CHAT_V1)
```

makes it clearer what the expected result is.

Comment (bparees) [reply tests/unit/llms/test_llm_loader.py:20]: Also for something like this:

1) separate PR from fixing the message template
2) would it be that hard to add error handling/exception raising in the llm_loader for this so we can have a meaningful test?

Comment (tisnik) [reply tests/unit/llms/test_llm_loader.py]: fixed the provider name @bparees 

Ok so is the expected behavior to raise an exception immediately the unknown provider is ... provided?

Comment (tisnik) [reply tests/unit/llms/test_llm_loader.py]: okay make sense, I'm on it

Comment (tisnik) [reply tests/unit/llms/test_llm_loader.py]: better now @bparees ? :)

**Comment by bparees** [comment]
/approve

fix the PR title and commit message to reflect that this is actually changing and i'll lgtm

**Comment by tisnik** [comment]
> /approve
> 
> fix the PR title and commit message to reflect that this is actually changing and i'll lgtm

Fixed. TY in advance

**Pull request merged**